src/app/related-search/related-search.component.spec.ts

```diff
@@ -79,7 +50,19 @@ describe('RelatedSearchComponent', () => {
     fixture.detectChanges();
   });
 
-  it('should create', () => {
+  it('should create an instance', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should display results if any, along with right keyword', () => {
+    let knowledgeService = fixture.debugElement.injector.get(KnowledgeapiService);
+    let compiled = fixture.debugElement.nativeElement;
+    if (component.results.length > 0) {
```

Comment:
@nikhilrayaprolu Like I said, the spy service gives errors since it is being called on the constructor function, I donot know how to prevent it, in the tutorial you have mentioned, the spy function requires a method to be called on